pkg/webhook/webhook_test.go

```diff
@@ -1160,3 +1163,215 @@ func TestMutateContainers(t *testing.T) {
 		})
 	}
 }
+
+func TestInjectProxyInitContainer(t *testing.T) {
+	proxyPort := int32(8080)
+	ProxySidecarVersion = "v1.0.0"
+	proxyInitContainer := corev1.Container{
+		Name:  ProxyInitContainerName,
+		Image: strings.Join([]string{ProxyInitImageRepository, ProxySidecarVersion}, ":"),
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add:  []corev1.Capability{"NET_ADMIN"},
+				Drop: []corev1.Capability{"ALL"},
+			},
+			Privileged: pointer.BoolPtr(true),
+			RunAsUser:  pointer.Int64Ptr(0),
+		},
+		Env: []corev1.EnvVar{{
+			Name:  ProxyPortEnvVar,
+			Value: strconv.FormatInt(int64(proxyPort), 10),
+		}},
+	}
+
+	tests := []struct {
+		name               string
+		containers         []corev1.Container
+		expectedContainers []corev1.Container
+	}{
+		{
+			name:               "no init containers",
+			containers:         []corev1.Container{},
+			expectedContainers: []corev1.Container{proxyInitContainer},
+		},
+		{
+			name:               "proxy init container manually injected",
+			containers:         []corev1.Container{proxyInitContainer},
+			expectedContainers: []corev1.Container{proxyInitContainer},
+		},
```

Comment:
Could we add a test with another existing init container to ensure we append to the existing list?